common/time-service/src/mock.rs

```diff
@@ -14,12 +14,25 @@ use std::{
     time::Duration,
 };
 
+/// TODO(philiphayes): Use `Duration::MAX` once it stabilizes.
+fn duration_max() -> Duration {
+    Duration::new(std::u64::MAX, 1_000_000_000 - 1)
+}
+
```

Comment:
Always wondered why this isn't stable yet.  Could you make it a const or static, so this is only executed once?

Reply:
Duration::new is not a const fn :'(

I'll add an `#[inline]` hint here for now : p

Reply:
What about static?  Since it's copyable.